uapi/include/uapi/dma.h

```diff
@@ -0,0 +1,166 @@
+ * this header declare various types that are DMA related. These types are
+ * used by both kernel and userspace in order to avoid any divergence.
+ *
+*/
+
+/**
+ * @enum gpdma_chan_trigger
+ *
+ * @brief type of DMA tranfer triggers request. This enum is made for bitfield manipulation,
+ * allowing multitple triggers at the same time if needed.
+ */
+typedef enum gpdma_chan_int {
+    GPDMA_INT_TC                    = 1, /**< DMA channel trigger on transfer complete */
+    GPDMA_INT_HT                    = 2, /**< DMA channel trigger on half transfer and transfer complete */
+    GPDMA_INT_ERROR                 = 4, /**< triggers on DMA transfer or config error, get status for complete information */
+} gpdma_chan_int_t;
+
+/**
+ * @brief generic DMA streams and channel status
+ *
+ * list DMA-generic status types that can be used by upper layer,
+ * whatever the DMA controller and driver is.
+ */
+typedef struct __attribute__((packed)) gpdma_chan_status {
+    uint32_t state: 3;
+    uint32_t half_reached: 1;
+    uint32_t completed: 1;
+    uint32_t reserved: 3;
+} gpdma_chan_status_t;
+
+static_assert(sizeof(gpdma_chan_status_t) == sizeof(uint8_t), "O,vamod gpdma_chan_status_t size");
+
+/**
+ * @enum gpdma_transfer_type
+ *
+ * @brief possible DMA transfer types
+ */
+typedef enum gpdma_transfer_type {
+    GPDMA_TRANSFER_MEMORY_TO_DEVICE = 0,
+    GPDMA_TRANSFER_DEVICE_TO_MEMORY = 1,
+    GPDMA_TRANSFER_MEMORY_TO_MEMORY = 2,
+    GPDMA_TRANSFER_DEVICE_TO_DEVICE = 3,
+} gpdma_transfer_type_t;
+
+/**
+ * @enum gpdma_transfer_mode
+ *
+ * @brief possible DMA transfer mode
+ *
+ * This enumerate behaves as a bitfield, allowing multiple values to be set
+ * at the same time.
+ */
+typedef enum gpdma_transfer_mode {
+    GPDMA_TRANSFER_MODE_INCREMENT_NONE = 0, /**< no increment at all of src and dest */
+    GPDMA_TRANSFER_MODE_INCREMENT_SRC  = 1, /**< increment src at each burst */
+    GPDMA_TRANSFER_MODE_INCREMENT_DEST = 2, /**< increment dest at each burst */
+} gpdma_transfer_mode_t;
+
+/**
+ * @enum gpdma_beat_len
+ *
+ * @brief possible DMA single manipulation data len (denoted beat)
+ * the beat len, associated to the transfer mode and the burst number defined how the
+ * DMA behaves.
+ *
+ * This defines the way burst works and impact the increment calculation. This
+ * is defined for both source and dest and may differ, allowing differenciated increment
+ */
+typedef enum gpdma_beat_len {
+    GPDMA_BEAT_LEN_BYTE = 0,     /**< data len to manipulate is a byte */
+    GPDMA_BEAT_LEN_HALFWORD = 1, /**< data len to manipulate is a half word */
+    GPDMA_BEAT_LEN_WORD = 2,     /**< data len to manipulate is a word */
+} gpdma_beat_len_t;
+
+/**
+ * @enum gpdma_priority
+ *
+ * @brief DMA stream priority in comparison with others
+ */
+typedef enum gpdma_priority {
+    GPDMA_PRIORITY_LOW    = 0,
+    GPDMA_PRIORITY_MEDIUM = 1,
+    GPDMA_PRIORITY_HIGH = 2,
+    GPDMA_PRIORITY_VERY_HIGH = 3,
+} gpdma_priority_t;
```

Comment:
Some of above structures are not used so far, correct?

Reply:
prio is a dts entry, yet used at driver level, this is only canonical here